### PR TITLE
Handle error responses with exception class

### PR DESCRIPTION
**Pull request by thomasjohansen**
- [ ] Bug fix
- [x] New feature
- [ ] BC breaks
- [ ] Deprecations
- [ ] Tests pass

Fixed tickets: none
License: MIT

Default Exception class added.

### CODE REVIEW

File: Middleware/EventDispatchMiddleware.php [hunk 1]
```diff
@@ -64,6 +64,16 @@ function (ResponseInterface $response) {
 
                         // Continue down the chain.
                         return $postTransactionEvent->getTransaction();
+                    },
+                    function (Exception $exception) {
```
Comment (florianpreusner) [inline Middleware/EventDispatchMiddleware.php]: You are not using any namespace, I think \Exception should be right.

**Comment by florianpreusner** [comment]
@thomasjohansen Why do you always create a new PR? You can always update your PR by pushing new commits...

**Comment by thomasjohansen** [comment]
Sorry - I'm new to this Github stuff.

**Comment by florianpreusner** [comment]
np. thanks for contributing.

**Comment by gabor-nagy-stylers** [comment]
because of this feature, the entire previous logic with option http_errors=true is messed up. no clientexception raising when response is not 200. so staying on 5.3. should not release such big modifications by minor version upgrade.

**Comment by florianpreusner** [comment]
You are totally right, @gabor-nagy-stylers. It was my mistake to release such changes as a minor version. I removed 5.4 and added the changes to a new major version.
I apologize for the inconvenience!

**Comment by gabor-nagy-stylers** [comment]
Thank you very much!

**Pull request merged**